src/kubernetes/Handlers/KubernetesResourceHandler.cs

```diff
@@ -1,46 +1,59 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 using System.Collections.Immutable;
-using Azure.Deployments.Extensibility.Providers.Kubernetes;
+using Azure.Deployments.Extensibility.Extensions.Kubernetes;
 using Bicep.Local.Extension.Protocol;
 using System.Diagnostics;
 using Azure.Deployments.Extensibility.Core.Json;
 using System.Text.Json.Nodes;
 using System.Text.Json;
-using ExtCore = Azure.Deployments.Extensibility.Core;
 using Json.More;
+using System.Formats.Asn1;
+using Microsoft.Extensions.DependencyInjection;
+using Azure.Deployments.Extensibility.AspNetCore;
+using Azure.Deployments.Extensibility.Extensions.Kubernetes.DependencyInjection;
+using V2Models = Azure.Deployments.Extensibility.Core.V2.Models;
+using Microsoft.AspNetCore.Http;
 
 namespace Bicep.Local.Extension.Kubernetes.Handlers;
 
 public partial class KubernetesResourceHandler : IGenericResourceHandler
 {
+    private static IExtension GetKubernetesExtension()
+    {
+        var serviceProvider = new ServiceCollection().AddKubernetesExtensionDispatcher().BuildServiceProvider();
+        var extensionDispatcher = serviceProvider.GetRequiredKeyedService<IExtensionDispatcher>("Kubernetes");
+
+        return extensionDispatcher.DispatchExtension("1.0.0");
+    }
+
     public async Task<LocalExtensibilityOperationResponse> Delete(Protocol.ResourceReference request, CancellationToken cancellationToken)
-        => Convert(await new KubernetesProvider().DeleteAsync(Convert(request), cancellationToken), request.Type, request.ApiVersion);
+        => Convert(await GetKubernetesExtension().DeleteResourceAsync(new(), Convert(request), cancellationToken), request.Type, request.ApiVersion);
```

Comment:
What should I pass in for `HttpContext` here?

Reply:
t`new DefaultHttpContext()`. It shouldn't cause any issue as the Kubernetes extension doesn't utilize `HttpContext`.